engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -8783,6 +8783,29 @@ object IR {
         override def message: String = s"The module $name does not exist."
       }
 
+      case class TypeDoesNotExist(
+        typeName: String,
+        moduleName: String
+      ) extends Reason {
+        override def message: String =
+          s"The type $typeName does not exist in module $moduleName"
+      }
+
+      case class SymbolsDoNotExist(
+        symbolNames: List[String],
+        moduleName: String
+      ) extends Reason {
+        override def message: String =
+          s"The symbols $symbolNames (modules or types) do not exist in module $moduleName."
+      }
+
+      case class NoSuchConstructors(
```

Comment:
Is there a reason for the plural form? `NoSuchConstructor` would work as well, right?

Reply:
Not really. It depends on whether we want to report all the unresolved symbols from one `from-import` statement, or just the first one. In this version, I tried to report all the unresolved symbols at once, but I guess reporting just the first one is enough.

Reply:
Reworked in 2a32eb0660d5322a157e9bc519731537589d55dc

Reply:
Feel free to report all the unresolved symbols, but that should be done by creating multiple instances of `NoSuchConstructor`?

Reply:
You are right. Done in 30c3420c32e058d72f3e2e7de8f350981e192242 - it now reports multiple errors at once by creating a list of `IR.Error.ImportError`.